src/body.rs

```diff
@@ -62,6 +63,47 @@ where
     }
 }
 
+#[cfg(feature = "hyper-body")]
+#[allow(unconditional_recursion)]
+impl<T> ::hyper::body::Payload for BoxBody<T>
+where
+    T: bytes::Buf + Send + IntoBuf + 'static,
+{
+    type Data = T;
+    type Error = ::hyper::Error;
+
+    fn poll_data(&mut self) -> Poll<Option<Self::Data>, Self::Error> {
+        ::hyper::body::Payload::poll_data(self)
+    }
+
+    fn poll_trailers(&mut self) -> Poll<Option<http::HeaderMap>, Self::Error> {
+        ::hyper::body::Payload::poll_trailers(self)
+    }
+
+    fn is_end_stream(&self) -> bool {
+        ::hyper::body::Payload::is_end_stream(self)
+    }
+}
+
+#[cfg(feature = "hyper-body")]
+impl Body for ::hyper::Body {
+    type Data = Bytes;
+
+    fn poll_data(&mut self) -> Poll<Option<Self::Data>, ::Status> {
+        match ::hyper::body::Payload::poll_data(self) {
+            Ok(Async::Ready(Some(chunk))) => Ok(Async::Ready(Some(chunk.into_bytes()))),
+            Ok(Async::Ready(None)) => unreachable!(),
```

Comment:
why is this unreachable?

Reply:
I realized this is wrong, it should be `Ok(Async::Ready(None))`